### PR TITLE
feat: added customer and delivery_date fields in pick list

### PR DESCRIPTION
**Pull request by nmimsnikhil**
task : https://bloomstack.com/desk#Form/Task/TASK-2020-01488
-added fields and fetched.

pick_list

### CODE REVIEW

File: erpnext/stock/doctype/pick_list/pick_list.json [hunk 1]
```diff
@@ -108,10 +110,27 @@
...
+  },
+  {
+   "depends_on": "eval:doc.purpose==='Delivery against Sales Order'",
+   "fieldname": "customer_name",
+   "fieldtype": "Link",
+   "in_list_view": 1,
+   "label": "Customer Name",
+   "options": "Customer",
+   "read_only": 1
+  },
```
Comment (Alchez) [inline erpnext/stock/doctype/pick_list/pick_list.json:122]: The customer link field already exists, did you mean to add a data field to show in the list view instead?

File: erpnext/stock/doctype/pick_list/pick_list.py [hunk 2]
```diff
@@ -22,10 +22,14 @@
...
 
 class PickList(Document):
+	def validate(self):
+		self.validate_delivery_date()
+
 	def before_save(self):
 		self.set_item_locations()
 
 	def before_submit(self):
+		self.set_item_locations()
```
Comment (Alchez) [inline erpnext/stock/doctype/pick_list/pick_list.py:32]: Why are we adding this one?

File: erpnext/stock/doctype/pick_list_item/pick_list_item.json [hunk 3]
```diff
@@ -166,10 +169,22 @@
    "fieldtype": "Data",
    "label": "Material Request Item",
    "read_only": 1
+  },
+  {
+   "fetch_from": "sales_order.customer",
+   "fieldname": "so_customer",
```
Comment (Alchez) [inline erpnext/stock/doctype/pick_list_item/pick_list_item.json:175]: Do we need this field?

File: erpnext/stock/doctype/pick_list/pick_list.py [hunk 4]
```diff
@@ -36,6 +40,14 @@ def before_submit(self):
 				continue
 			frappe.throw(_('For item {0} at row {1}, count of serial numbers does not match with the picked quantity')
 				.format(frappe.bold(item.item_code), frappe.bold(item.idx)))
+	
+	def validate_delivery_date(self):
+		so = frappe.get_doc("Sales Order", self.locations[0].get("sales_order")).as_dict()
```
Comment (Alchez) [inline erpnext/stock/doctype/pick_list/pick_list.py:45]: Why only the first row? Also, check if there's a value in the first place, and maybe an additional check for the Pick List type to be "Delivery against Sales Order".

File: erpnext/stock/doctype/pick_list/pick_list.py [hunk 5]
```diff
@@ -36,6 +40,14 @@ def before_submit(self):
...
+		item = so.get("items")
+		for i in range(len(so.get("items"))-1):
+			if item[i].delivery_date != item[i+1].delivery_date:
+				return
+		self.delivery_date = item[0].delivery_date
```
Comment (Alchez) [inline erpnext/stock/doctype/pick_list/pick_list.py:50]: Maybe find the max date, and set that instead of returning

**Review by Alchez — CHANGES_REQUESTED** [review]

File: erpnext/stock/doctype/pick_list/pick_list.json (hunk 1 above)
Comment (nmimsnikhil) [reply erpnext/stock/doctype/pick_list/pick_list.json:122]: yes i wanted to show customer_name in list view

File: erpnext/stock/doctype/pick_list/pick_list.py (hunk 2 above)
Comment (nmimsnikhil) [reply erpnext/stock/doctype/pick_list/pick_list.py:32]: by mistake i'll change and add a new commit

File: erpnext/stock/doctype/pick_list/pick_list.json (hunk 1 above)
Comment (Alchez) [reply erpnext/stock/doctype/pick_list/pick_list.json:122]: Change the fieldtype to `Data` then and fetch it from the linked customer (use the Fetch From field), we don't need 2 link fields for the same data.

**Comment by Alchez** [comment]
@nmimsnikhil, please make this to the `staging` branch instead

**Comment by Alchez** [comment]
Moved to #490.

**Pull request closed without merging**